packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -862,7 +860,7 @@ export class LocalFluidDataStoreContextBase extends FluidDataStoreContext {
         );
 
         this.snapshotTree = props.snapshotTree;
-        this.isRootDataStore = props.isRootDataStore;
+        this.isRootDataStore = props.isRootDataStore ?? false;
```

Comment:
It feels like that (usage of default) can be pushed up the stack, i.e. props.isRootDataStore  to be boolean and force callers to make a choice. I have not traced it (if it's easy and makes sense to do), but this might help find cases where callers should reconsider

Reply:
Created an issue to track this - https://github.com/microsoft/FluidFramework/issues/9355. We should fix this in main.